src/components/map/Map.tsx

```diff
@@ -122,14 +122,24 @@ class LeafletMap extends React.Component<IMapProps, IMapState> {
         this.getMap().setView(latLng, 17);
     }
 
+    private setDisabledNodeIds = (links: IRoutePathLink[]) => {
```

Comment:
Why have setDisabledNodeIds at Map.tsx? 

What about having a method at RouteLayer

setRouteOnTop() {
    setHoveredPolylines();
    setDisabledNodeIds();
}

Reply:
Also is there a way to go without forceUpdate()?

Reply:
Yes. It is no longer needed after moving setDisabledNodeIds to RouteLayer.